akka-actor-typed/src/main/scala/akka/actor/typed/internal/routing/GroupRouterImpl.scala

```diff
@@ -24,25 +23,60 @@ private[akka] final case class GroupRouterBuilder[T] private[akka] (
     with scaladsl.GroupRouter[T] {
 
   // deferred creation of the actual router
-  def apply(ctx: TypedActorContext[T]): Behavior[T] = new GroupRouterImpl[T](ctx.asScala, key, logicFactory())
+  def apply(ctx: TypedActorContext[T]): Behavior[T] = new InitialGroupRouterImpl[T](ctx.asScala, key, logicFactory())
 
   def withRandomRouting(): GroupRouterBuilder[T] = copy(logicFactory = () => new RoutingLogics.RandomLogic[T]())
 
   def withRoundRobinRouting(): GroupRouterBuilder[T] = copy(logicFactory = () => new RoutingLogics.RoundRobinLogic[T])
 
 }
 
+/**
+ * INTERNAL API
+ *
+ * Starting behavior for a group router before it got a first listing back from the receptionist
+ */
+@InternalApi
+private final class InitialGroupRouterImpl[T](
+    ctx: ActorContext[T],
+    serviceKey: ServiceKey[T],
+    routingLogic: RoutingLogic[T])
+    extends AbstractBehavior[T] {
+
+  // casting trix to avoid having to wrap incoming messages - note that this will cause problems if intercepting
+  // messages to a router
+  ctx.system.receptionist ! Receptionist.Subscribe(serviceKey, ctx.self.unsafeUpcast[Any].narrow[Receptionist.Listing])
+
+  private val stash = StashBuffer[T](capacity = 10000)
```

Comment:
Could this be passed in with this as a default? 10000 is quite a lot if they don't start for a while

Reply:
It's not about starting for a while it is about getting a reply from the receptionist, which is local, so should be very quick in all reasonable cases. Let's see if it ever is an issue before we turn it into a config knob